astropy/wcs/src/wcslib_wrap.c

```diff
@@ -3997,6 +4348,24 @@ PyWcsprm_set_zsource(
 }
 
 
+ /*@null@*/ static PyObject*
+PyWcsprm_get_unit_scaling(
+    PyWcsprm* self,
+    /*@unused@*/ void* closure) {
```

Comment:
Maybe add some documentation (for the developers) here: when does it return None and when an array of let's say 1s? What is the difference between `None` and `[1, 1]`?

Reply:
Would it be more consistent to return a list of `[None, None]` instead of `[1, 1]` or when only some axes need conversion then `[None, None, 1e3]`?

Reply:
Given that this is private API I would rather keep it simple and just mirror what is happening in the C code since it's just to examine unit_scaling.

Either we use the unit conversion path in the code in which case we have an array of floats or we don't use the conversions in which case we return None. We couldn't easily return a mixed double/None array without using an object array or a list, and it's not what's happening in the C code.

Would you be happy for me to leave as is if I just add a docstring?